.github/workflows/ci.yml

```diff
@@ -0,0 +1,70 @@
+name: CI
+
+on: ['pull_request']
+
+jobs:
+  test:
+    name: 🧪 Test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      
+      - name: 🔧 Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+          cache: 'yarn'
+          
+      - name: 📦 Install dependencies
+        run: yarn install --frozen-lockfile
+        
+      - name: 🧪 Run tests
+        run: yarn test:ci
+        
+      - name: 📊 Upload test results
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          name: test-results
+          path: |
+            junit.xml
+            coverage/
+
+  lint:
+    name: 🔍 Lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      
+      - name: 🔧 Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+          cache: 'yarn'
+          
+      - name: 📦 Install dependencies
+        run: yarn install --frozen-lockfile
+        
+      - name: 🔍 Run ESLint
+        run: yarn lint
+        
+      - name: 💅 Check formatting
+        run: yarn format:check
+
+  typecheck:
+    name: ʦ Type check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      
+      - name: 🔧 Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+          cache: 'yarn'
+          
+      - name: 📦 Install dependencies
+        run: yarn install --frozen-lockfile
+        
+      - name: ʦ Type check
+        run: yarn tsc --noEmit
```

Comment:
Nit:
The linter is complaining about the missing newline at the end of the doc.